### DISCUSSION TITLE
React SDK - passing context and initialization

### DISCUSSION THREAD

**Discussion by IrinaSouth** [featurehub-io/featurehub-javascript-sdk#171] [Q&A]
1. Can the following be achieved in the React SDK?
```
await fhContext.userKey('ideally-unique-id')
.country(StrategyAttributeCountryName.NewZealand)
.device(StrategyAttributeDeviceName.Browser)
.platform(StrategyAttributePlatformName.Android)
.version('1.2.0')
.build();
```
2. Is there is a possibility to delay the SDK initialization, for example after user logs in and gets set values?

**Comment by rvowles** [comment]
In answer to (1), the context and config that has been initialized within the <FeatureHub> class in the ReactSDK is set globally in the window context. It happens at this location https://github.com/featurehub-io/featurehub-javascript-sdk/blob/main/featurehub-react-sdk/src/components/FeatureHub.tsx#L64 

Because it is set globally for the window, you can then grab it in your code as FeatureHub.context() - see here https://github.com/featurehub-io/featurehub-javascript-sdk/blob/main/featurehub-javascript-client-sdk/app/index.ts#L36 and do the code from (1) above.

To answer (2), in React, SDK initialization happens when you wrap the <FeatureHub> class around your code, it assumes from that point down you want access to features, so _currently_ the answer is that you would have to push that class further down or create your own variant of this one file and make it do what you wish, the React SDK is just window-dressing on top of the client-sdk. However, we would be happy to add a "delayInitialization" field (or similar) to make it do this for you. If that is of interest, please feel free to raise an issue in the Issues section and we'll see what we can do for you!
